### PR TITLE
update: Updates the HttpClientDefault set on Beagle helpers and on DOCS v1.7

### PR DESCRIPTION
**Pull request by carlossteinzup**

### CODE REVIEW

**Review by matheusribeirozup — APPROVED** [review]

File: content/en/resources/customization/beagle-for-android/network-client.md [hunk 1]
```diff
@@ -167,13 +166,15 @@ import kotlinx.coroutines.cancel
...
+import java.net.URI
 
 typealias OnSuccess = (responseData: ResponseData) -> Unit
 typealias OnError = (responseData: ResponseData) -> Unit
 
+@BeagleComponent
 class HttpClientDefault : HttpClient, CoroutineScope {
 
-    private val job = Job()
+        private val job = Job()
```
Comment (matheusribeirozup) [inline content/en/resources/customization/beagle-for-android/network-client.md:177]: Unnecessary change

File: content/en/resources/customization/beagle-for-android/network-client.md [hunk 2]
```diff
@@ -212,18 +216,20 @@ class HttpClientDefault : HttpClient, CoroutineScope {
...
             throw BeagleApiException(ResponseData(-1, data = byteArrayOf()), request)
         }
 
-        request.headers.forEach {
+        request.httpAdditionalData.headers?.forEach {
             urlConnection.setRequestProperty(it.key, it.value)
         }
 
-        addRequestMethod(urlConnection, request.method)
+        addRequestMethod(urlConnection, request.httpAdditionalData.method!!)
```
Comment (matheusribeirozup) [inline content/en/resources/customization/beagle-for-android/network-client.md:229]:
```suggestion
        request.httpAdditionalData.method?.let { method ->
            addRequestMethod(urlConnection, method)
        }
```

File: content/en/resources/customization/beagle-for-android/network-client.md [hunk 3]
```diff
@@ -212,18 +216,20 @@ class HttpClientDefault : HttpClient, CoroutineScope {
...
 
-        if (request.body != null) {
+        val body = request.httpAdditionalData.body
+        if (body != null) {
```
Comment (matheusribeirozup) [inline content/en/resources/customization/beagle-for-android/network-client.md:232]:
```suggestion
        request.httpAdditionalData.body?.run {
```

File: content/en/resources/customization/beagle-for-android/network-client.md [hunk 4]
```diff
@@ -236,13 +242,16 @@ class HttpClientDefault : HttpClient, CoroutineScope {
...
+    ): BeagleApiException {
         val response = urlConnection.getSafeError() ?: byteArrayOf()
         val statusCode = urlConnection.getSafeResponseCode()
         val statusText = urlConnection.getSafeResponseMessage()
-        val responseData = ResponseData(statusCode = statusCode,
-            data = response, statusText = statusText)
-
+        val responseData = ResponseData( statusCode = statusCode,
+            data = response, statusText = statusText
+        )
```
Comment (matheusribeirozup) [inline content/en/resources/customization/beagle-for-android/network-client.md:254]:
```suggestion
        val responseData = ResponseData(
            statusCode = statusCode,
            data = response,
            statusText = statusText,
        )
```

File: content/pt/resources/customization/beagle-for-android/network-client.md [hunk 5]
```diff
@@ -169,14 +168,15 @@ import kotlinx.coroutines.cancel
...
+import java.net.URI
 
 typealias OnSuccess = (responseData: ResponseData) -> Unit
 typealias OnError = (responseData: ResponseData) -> Unit
 
+@BeagleComponent
 class HttpClientDefault : HttpClient, CoroutineScope {
 
-    private val job = Job()
+        private val job = Job()
```
Comment (matheusribeirozup) [inline content/pt/resources/customization/beagle-for-android/network-client.md:179]: Unnecessary change

File: content/pt/resources/customization/beagle-for-android/network-client.md [hunk 6]
```diff
@@ -215,18 +218,20 @@ class HttpClientDefault : HttpClient, CoroutineScope {
...
             throw BeagleApiException(ResponseData(-1, data = byteArrayOf()), request)
         }
 
-        request.headers.forEach {
+        request.httpAdditionalData.headers?.forEach {
             urlConnection.setRequestProperty(it.key, it.value)
         }
 
-        addRequestMethod(urlConnection, request.method)
+        addRequestMethod(urlConnection, request.httpAdditionalData.method!!)
```
Comment (matheusribeirozup) [inline content/pt/resources/customization/beagle-for-android/network-client.md:231]:
```suggestion
        request.httpAdditionalData.method?.let { method ->
            addRequestMethod(urlConnection, method)
        }
```

File: content/pt/resources/customization/beagle-for-android/network-client.md [hunk 7]
```diff
@@ -215,18 +218,20 @@ class HttpClientDefault : HttpClient, CoroutineScope {
...
 
-        if (request.body != null) {
+        val body = request.httpAdditionalData.body
+        if (body != null) {
```
Comment (matheusribeirozup) [inline content/pt/resources/customization/beagle-for-android/network-client.md:234]:
```suggestion
        request.httpAdditionalData.body?.run {
```

File: content/pt/resources/customization/beagle-for-android/network-client.md [hunk 8]
```diff
@@ -239,13 +244,16 @@ class HttpClientDefault : HttpClient, CoroutineScope {
...
+    ): BeagleApiException {
         val response = urlConnection.getSafeError() ?: byteArrayOf()
         val statusCode = urlConnection.getSafeResponseCode()
         val statusText = urlConnection.getSafeResponseMessage()
-        val responseData = ResponseData(statusCode = statusCode,
-            data = response, statusText = statusText)
-
+        val responseData = ResponseData( statusCode = statusCode,
+            data = response, statusText = statusText
+        )
```
Comment (matheusribeirozup) [inline content/pt/resources/customization/beagle-for-android/network-client.md:256]:
```suggestion
        val responseData = ResponseData(
            statusCode = statusCode,
            data = response,
            statusText = statusText,
        )
```

File: content/en/resources/customization/beagle-for-android/network-client.md [hunk 9]
```diff
@@ -154,7 +154,6 @@ HttpClientDefault class defines how the services requests are configured. To use
 This configuration is long, so copy and paste the class below. You may modify it later.
 
 ```kotlin
-import br.com.zup.beagle.android.annotation.BeagleComponent
```
Comment (matheusribeirozup) [inline content/en/resources/customization/beagle-for-android/network-client.md:157]: Should this import be removed?

**Review by matheusribeirozup — COMMENTED** [review]
Please consider making changes

File: content/en/resources/customization/beagle-for-android/network-client.md (hunk 9 above)
Comment (carlossteinzup) [reply content/en/resources/customization/beagle-for-android/network-client.md:157]: It shouldn't on the doc. It was removed on the beagle helpers since we dont use the anotation there. Thanks!!

File: content/en/resources/customization/beagle-for-android/network-client.md [hunk 10]
```diff
@@ -154,7 +154,6 @@ HttpClientDefault class defines how the services requests are configured. To use
 This configuration is long, so copy and paste the class below. You may modify it later.
 
 ```kotlin
```
Comment (carlossteinzup) [inline content/en/resources/customization/beagle-for-android/network-client.md:156]:
```suggestion
```kotlin
import br.com.zup.beagle.android.annotation.BeagleComponent
```

File: content/en/resources/customization/beagle-for-android/network-client.md (hunk 1 above)
Comment (carlossteinzup) [inline content/en/resources/customization/beagle-for-android/network-client.md:178]:
```suggestion
private val job = Job()
```

Comment (carlossteinzup) [inline content/en/resources/customization/beagle-for-android/network-client.md:178]:
```suggestion
   private val job = Job()
```

Comment (carlossteinzup) [inline content/en/resources/customization/beagle-for-android/network-client.md:178]:
```suggestion
   private val job = Job()
```

File: content/pt/resources/customization/beagle-for-android/network-client.md (hunk 5 above)
Comment (carlossteinzup) [inline content/pt/resources/customization/beagle-for-android/network-client.md:179]:
```suggestion
   private val job = Job()
```

**Review by matheusribeirozup — APPROVED** [review]
